src/components/organisms/TestConfigurationForm/molecules/Revision.tsx

```diff
@@ -0,0 +1,34 @@
+import {useState} from 'react';
+import {useFirstMountState} from 'react-use';
+
+import {FormItem} from '@custom-antd';
+
+import {Branch, Commit, RevisionSwitcher} from '.';
+import {StyledFormSpace} from '../TestConfigurationForm.styled';
+
+const Revision = () => {
+  const [switcherValue, setSwitcherValue] = useState<string | number>('branch');
+
+  const isFirst = useFirstMountState();
+
+  return (
+    <FormItem noStyle shouldUpdate>
+      {({getFieldValue}) => {
+        const commit = getFieldValue('commit');
+
+        if (isFirst) {
+          setSwitcherValue(commit ? 'commit' : 'branch');
+        }
+
+        return (
+          <StyledFormSpace size={24} direction="vertical">
+            <RevisionSwitcher value={switcherValue} onChange={setSwitcherValue} />
```

Comment:
By changing to `Revision`, I didn't mean component name - I meant, that we could get rid of `Commit`, `Branch`, and `RevisionSwitcher` and just have `Revision` - simple text input with `Git revision` label.

It doesn't matter whether the user will provide a branch name or commit hash. Git commands will operate on any:

```
git checkout main
git checkout HEAD
git checkout 23e55a1449cf0bcc4b8e6cd7fbcdfbd97d2e662c
```

Reply:
Just tested, and for `commit` property at the moment you may pass either commit hash or branch name and it works, so it behave as `revision` would :)

Reply:
@vsukhin, what do you think about `revision` approach?

Reply:
Oh, I see. I am not sure about it. Probably need to discuss with the team